ddp-workspace/projects/ddp-sdk/src/lib/components/languageSelector.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { StudyLanguage } from "../models/studyLanguage";
+import { LanguageServiceAgent } from "../services/serviceAgents/languageServiceAgent.service";
+import { isNullOrUndefined } from "util";
+import { TranslateService } from "@ngx-translate/core";
+
+@Component({
+  selector: 'ddp-language-selector',
+  template: `
+    <div [hidden]="!loaded">
+      <button class="SimpleButton" [ngClass]="{'SimpleButton--Scrolled': isScrolled}" *ngIf="currentLanguage !== null && currentLanguage !== undefined" [matMenuTriggerFor]="menu">
+        <svg class="ddp-globe" height="24px" width="24px" aria-live="title">
+          <title id="title" [lang]="currentLanguage.languageCode" translate>Toolkit.Header.LanguageSelection</title>
+          <use href="assets/images/globe.svg#Language-Selector-3"></use>
+        </svg>
+        <span class="ddp-current-language">{{currentLanguage.displayName}}</span>
+        <mat-icon class="ddp-dropdown-arrow">arrow_drop_down</mat-icon>
+      </button>
+      
+      <mat-menu #menu="matMenu">
+        <button mat-menu-item *ngFor="let lang of getUnselectedLanguages()" (click)="changeLanguage(lang)">{{lang.displayName}}</button>
+      </mat-menu>
+    </div>
+  `
+})
+export class LanguageSelectorComponent implements OnInit {
+  @Input() studyGuid: string;
+  @Input() isScrolled: boolean;
+  public loaded: boolean;
+  private studyLanguages: Array<StudyLanguage>;
+  private currentLanguage: StudyLanguage;
+
+  constructor (
+    private serviceAgent: LanguageServiceAgent,
+    private translateService: TranslateService
+  ) { }
+
+  public ngOnInit(): void {
+    this.serviceAgent.getConfiguredLanguages(this.studyGuid).subscribe(x => {
+      if (x) {
+        console.log('got configured languages: ' + JSON.stringify(x));
+        //Only use language selector if multiple languages are configured!
+        if (x.length > 1) {
+          this.studyLanguages = x;
+          this.translateService.addLangs(this.studyLanguages.map(x => x.languageCode));
+          if (this.findCurrentLanguage()) {
+            this.loaded = true;
+          }
+        }
+      } else {
+        console.error('Error: no configured language list was returned.');
+      }
+    });
+  }
+
+  private getUnselectedLanguages(): Array<StudyLanguage> {
+    if (this.studyLanguages !== null  && this.studyLanguages !== undefined) {
+      return this.studyLanguages.filter(elem => elem !== this.currentLanguage);
+    }
+   return null;
+  }
+
+  private changeLanguage(lang: StudyLanguage): void {
+    this.currentLanguage = lang;
+    this.translateService.use(lang.languageCode);
+    sessionStorage.setItem('studyLanguage', lang.languageCode);
```

Comment:
I actually considered making this change.  I went back and forth on whether to make it, mostly because I was worried there was a case I had thought of earlier and that was why I had planned on doing session storage.  But I think local storage is the right call here.  In particular, apparently session storage, at least in Chrome, is tab-specific.